src/components/SlateEditor/plugins/link/Link.tsx

```diff
@@ -70,160 +62,128 @@ export interface Model {
+        editor,
+        {
+          type: TYPE_CONTENT_LINK,
+          data: { ...data, resource: "content-link" },
+          isFirstEdit: false,
+        },
+        {
+          at: path,
+          match: (node) => Element.isElement(node) && (node.type === TYPE_LINK || node.type === TYPE_CONTENT_LINK),
+        },
+      );
+    }
+    setEditMode(false);
+    setTimeout(() => {
+      Transforms.select(editor, Path.next(ReactEditor.findPath(editor, element)));
+      Transforms.collapse(editor, { edge: "start" });
+      ReactEditor.focus(editor);
+    }, 0);
+  };
 
   return (
-    <Modal defaultOpen={startOpen.current} open={editMode} onOpenChange={toggleEditMode}>
-      <StyledLink {...attributes} href={model?.href} ref={linkRef}>
-        <InlineBugfix />
-        {children}
-        <InlineBugfix />
-        {model && (
-          <Portal asChild>
-            <StyledLinkMenu
-              top={top}
-              left={left}
-              data-visible={selected}
-              onMouseDown={(e) => {
-                e.preventDefault();
-                e.stopPropagation();
-                setEditorSelection(editor.selection);
-              }}
-            >
-              <ModalTrigger>
-                <Button variant="link">{t("form.content.link.change")}</Button>
-              </ModalTrigger>{" "}
-              | {t("form.content.link.goTo")}{" "}
-              <a href={model?.href} target="_blank" rel="noopener noreferrer">
-                {model?.href}
-              </a>
-            </StyledLinkMenu>
-          </Portal>
-        )}
-      </StyledLink>
-      <ModalContent
-        onEscapeKeyDown={(e) => e.stopPropagation()}
-        onCloseAutoFocus={(e) => {
-          if (!model?.href) {
-            handleRemove();
+    <DialogRoot open={editMode} onOpenChange={(details) => toggleEditMode(details.open)}>
+      <PopoverRoot
+        modal={false}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus={false}
+        onOpenChange={(details) => {
+          if (!details.open) {
+            ReactEditor.focus(editor);
           }
-          e.preventDefault();
-          focusEditor();
         }}
       >
-        {model && (
-          <EditLink
-            editor={editor}
-            element={element}
-            model={model}
-            closeEditMode={() => toggleEditMode(false)}
-            handleRemove={handleRemove}
-          />
-        )}
-      </ModalContent>
-    </Modal>
+        <PopoverTrigger asChild consumeCss>
+          <a {...attributes} href={linkData.href} ref={linkRef}>
+            <InlineBugfix />
+            {children}
+            <InlineBugfix />
+          </a>
+        </PopoverTrigger>
+        <Portal>
+          <StyledPopoverContent>
+            <DialogTrigger asChild>
+              <Button variant="link">{t("form.content.link.change")}</Button>
+            </DialogTrigger>{" "}
+            | {t("form.content.link.goTo")}{" "}
```

Comment:
Liker ikkje heilt disse for å få inn mellomrom. Antar det er editoren som har gjort det for deg.